lib/Db/MessageMapper.php

```diff
@@ -824,16 +848,19 @@ public function findByIds(string $userId, array $ids): array {
 			return [];
 		}
 		$qb = $this->db->getQueryBuilder();
+		$results = [];
+		foreach (array_chunk($ids, 1000) as $chunk) {
+			$select = $qb
+				->select('*')
+				->from($this->getTableName())
+				->where(
+					$qb->expr()->in('id', $qb->createNamedParameter($chunk, IQueryBuilder::PARAM_INT_ARRAY), IQueryBuilder::PARAM_INT_ARRAY)
+				)
+				->orderBy('sent_at', 'desc');
 
-		$select = $qb
-			->select('*')
-			->from($this->getTableName())
-			->where(
-				$qb->expr()->in('id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_INT_ARRAY), IQueryBuilder::PARAM_INT_ARRAY)
-			)
-			->orderBy('sent_at', 'desc');
-
-		return $this->findRelatedData($this->findEntities($select), $userId);
+			$results = array_merge($results, $this->findRelatedData($this->findEntities($select), $userId));
```

Comment:
array_merge inside loops is not recommended: https://github.com/kalessil/phpinspectionsea/blob/master/docs/performance.md#slow-array-function-used-in-loop

Reply:
Very interesting, good to know!